.github/workflows/ci.yml

```diff
@@ -28,6 +28,30 @@ jobs:
 
       - run: make test
 
+  e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+
+      - name: Install Go dependencies
+        run: |
+          go install github.com/onsi/ginkgo/v2/ginkgo
+          go install sigs.k8s.io/kwok/cmd/{kwok,kwokctl}@v0.3.0
```

Comment:
kwok and kind both required for the same e2e test? but you only use kind cluster provider below?